src/main/java/com/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,133 @@
+      Optional<User> user = userRepository.findById(id);
+        if(user.isPresent()){
+            userRepository.deleteById(id);
+            return new ResponseEntity<>( "User has been successfully removed", HttpStatus.OK);
+        }else{
+           throw new UserNotFoundException();
+        }
+
+    }
+
+    /**
+     * update user in the database with the received user
+     * @param user to be updated
+     * @param id to search for user to update
+     * @exception UserUsernameRepeatedException when the username is already in the database.
+     * @exception UserNotFoundException when user was not found
+     * @return user updated
+     */
+    @PutMapping("/{id}")
+    public ResponseEntity<Object> updateUser(@RequestBody User user, @PathVariable Long id) {
+        User addUser = userRepository.findById(id).orElseThrow(UserNotFoundException::new);
+
+        if(!addUser.getUsername().equalsIgnoreCase(user.getUsername())){
+            if(userRepository.findByUsername(user.getUsername()).isPresent()) {
+                throw new UserUsernameRepeatedException();
+            }
+            addUser.setUsername(user.getUsername());
+        }
+        addUser.setBirthdate(user.getBirthdate());
+        addUser.setName(user.getName());
+        return new ResponseEntity<>(userRepository.save(addUser), HttpStatus.OK);
+    }
+
+    @GetMapping
+    public ResponseEntity<Object> findAll() {
+        return new ResponseEntity<>( userRepository.findAll(), HttpStatus.OK);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Object> findOne(@PathVariable Long id) {
+        return new ResponseEntity<>(userRepository.findById(id).orElseThrow(UserNotFoundException::new), HttpStatus.OK);
+    }
+
+    /**
+     * add the received book to the list books
+     * @param book  book to be added
+     * @param id To find the user who owns the list
+     * @exception UserNotFoundException when user was not found
+     * @exception BookNotFoundException when the book was not found
+     * @return user updated
+     */
+    @PostMapping("addbook/{id}")
+    public ResponseEntity<Object> addBook(@RequestBody Book book, @PathVariable Long id) {
+        User user = userRepository.findById(id).orElseThrow(UserNotFoundException::new);
+        Optional<Book> addBook = bookRepository.findById(book.getId());
+        if(!addBook.isPresent()){
+            throw new BookNotFoundException();
+        }
+        user.addBook(addBook.get());
+        return new ResponseEntity<>( userRepository.save(user), HttpStatus.OK);
+    }
+
+    /**
+     * Identical to the add endpoint but it removes the book
+     */
+    @PostMapping("removebook/{id}")
+    public ResponseEntity<Object> removeBook(@RequestBody Book book, @PathVariable Long id) {
+        User user = userRepository.findById(id).orElseThrow(UserNotFoundException::new);
+        Optional<Book> removeBook = bookRepository.findById(book.getId());
```

Comment:
You could use `.orElseThrow(BookNotFoundException::new)` and we would avoid the if below